pkg/operator/sync.go

```diff
@@ -244,6 +246,96 @@ func (optr *Operator) syncRenderConfig(_ *renderConfig) error {
 	}
 
+	// handle image registry certificates.
+	// parse these, add them to ctrlcfgspec and then handle these in the daemon write to disk function
+	cfg, err := optr.configClient.ConfigV1().Images().Get(context.TODO(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	imgRegistryUsrData := []v1.ImageRegistryBundle{}
+	if cfg.Spec.AdditionalTrustedCA.Name != "" {
+		klog.Infof("got image.config.openshift.io/cluster, additionalCA: %s", cfg.Spec.AdditionalTrustedCA.Name)
+		cm, err := optr.clusterCmLister.ConfigMaps("openshift-config").Get(cfg.Spec.AdditionalTrustedCA.Name)
+		if err != nil {
+			klog.Warningf("could not find configmap specified in image.config.openshift.io/cluster with the name %s", cfg.Spec.AdditionalTrustedCA.Name)
+		}
+		for key := range cm.Data { // default to binay data if it is there, else use data
+			if bd, bdok := cm.BinaryData[key]; bdok {
+				imgRegistryUsrData = append(imgRegistryUsrData, v1.ImageRegistryBundle{
+					File: key,
+					Data: bd,
+				})
+			} else if d, dok := cm.Data[key]; dok {
+				raw, err := base64.StdEncoding.DecodeString(d)
+				if err != nil {
+					imgRegistryUsrData = append(imgRegistryUsrData, v1.ImageRegistryBundle{
+						File: key,
+						Data: []byte(d),
+					})
+				} else {
+					imgRegistryUsrData = append(imgRegistryUsrData, v1.ImageRegistryBundle{
+						File: key,
+						Data: raw,
+					})
+				}
+			}
+		}
+	}
+
+	imgRegistryData := []v1.ImageRegistryBundle{}
+	cm, err := optr.clusterCmLister.ConfigMaps("openshift-config-managed").Get("image-registry-ca")
+	if err != nil {
+		klog.Warningf("could not find configmap image-registry-ca, cluster-image-registry-operator might not be installed.")
+	} else {
+		for key, _ := range cm.Data { // default to binay data if it is there, else use data
+			klog.Infof("mandatory data for img registry, key: %s data: %s", key, cm.Data[key])
+			if bd, bdok := cm.BinaryData[key]; bdok {
+				imgRegistryData = append(imgRegistryData, v1.ImageRegistryBundle{
+					File: key,
+					Data: bd,
+				})
+			} else if d, dok := cm.Data[key]; dok {
+				raw, err := base64.StdEncoding.DecodeString(d)
+				if err != nil {
+					imgRegistryData = append(imgRegistryData, v1.ImageRegistryBundle{
+						File: key,
+						Data: []byte(d),
+					})
+				} else {
+					imgRegistryData = append(imgRegistryData, v1.ImageRegistryBundle{
+						File: key,
+						Data: raw,
+					})
+				}
+			}
+		}
+	}
+
+	mergedData := append(imgRegistryData, imgRegistryUsrData...)
+	caData := make(map[string]string, len(mergedData))
+	for _, CA := range mergedData {
+		caData[CA.File] = string(CA.Data)
+	}
+
+	err = optr.kubeClient.CoreV1().ConfigMaps("openshift-config-managed").Delete(context.TODO(), "merged-trusted-image-registry-cas", metav1.DeleteOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	_, err = optr.kubeClient.CoreV1().ConfigMaps("openshift-config-managed").Create(
```

Comment:
so who consumes this merged ca configmap? I thought the image registry operator was still doing a merge as well? Or is that a separate thing?

Reply:
the merge is for:


```
There are also consumers that don't have access to the host file system: cluster-openshift-apiserver-operator needs a merged config map with all certificate authorities for image registries, so that openshift-apiserver can access registries.

For them, the mechanism should create a merged config map in openshift-config-managed.
```